tools/common/cache.cs

```diff
@@ -186,12 +186,12 @@ public unsafe static bool CompareStreams (Stream astream, Stream bstream, bool i
 		} while (true);
 	}
 	
-	string GetArgumentsForCacheData ()
+	string GetArgumentsForCacheData (Application app)
```

Comment:
Are you likely to use any other properties of `app`? If not, why not pass in `ProductConstants`?

Reply:
> Are you likely to use any other properties of `app`?

Logic says no, experience says yes... somehow most places end up using many `app` properties.